spring-data-mongodb/src/main/java/org/springframework/data/mongodb/core/query/NearQuery.java

```diff
@@ -420,25 +426,54 @@ public DBObject toDBObject() {
 		}
 
 		if (maxDistance != null) {
-			dbObject.put("maxDistance", maxDistance.getNormalizedValue());
+			dbObject.put("maxDistance", getDistanceValueInRadiantsOrMeters(maxDistance));
 		}
 
 		if (minDistance != null) {
-			dbObject.put("minDistance", minDistance.getNormalizedValue());
+			dbObject.put("minDistance", getDistanceValueInRadiantsOrMeters(minDistance));
 		}
 
 		if (metric != null) {
-			dbObject.put("distanceMultiplier", metric.getMultiplier());
+			dbObject.put(
+					"distanceMultiplier",
+					usesMetricSystem() ? BigDecimal.ONE.divide(getMetricSystemNormalizer(metric), PRECISION + 3,
```

Comment:
to achieve a calculation that is accurate to 0.3 meters one needs to use `BigDecimal` with the given precision